llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -9549,6 +9488,21 @@ LoopVectorizationPlanner::tryToBuildVPlanWithVPRecipes(VFRange &Range) {
   // Adjust the recipes for any inloop reductions.
   adjustRecipesForReductions(Plan, RecipeBuilder, Range.Start);
 
+  // Transform recipes to abstract recipes if it is leagal and beneficial and
+  // clamp the range for better cost estimation.
+  // TODO: Enable following transform when the EVL-version of extended-reduction
+  // and mulacc-reduction are implemented.
+  if (!CM.foldTailWithEVL()) {
```

Comment:
Why do we need to special case this for EVL? Shouldn't the cost-model tell us that the combined reductions aren't profitable?